### PR TITLE
ci(advanced-api/automatic-vendor-sharing sample):add(ci pipeline for e2e test + cache for e2e)

### PR DESCRIPTION
**Pull request by PeaceDate**
Added CI pipeline in Github Actions for running e2e Cypress tests for module federation examples samples.

E2E test running only by affected samples. Affected samples found by `lerna ls` command with `--since=origin/master` key, it gives us possibility to compare samples in current branch and origin/master branch and realise what actually was changed. 

Check fork is up to date - checks if the fork master branch is up to date with the origin master branch, if not pipeline failed

Pipeline structure:

First pipe:

- [x] Stop previous workflows
- [x] Check that fork is up to date with the origin
- [x]  Setup matrix for each sample
- [x] Run Cypress e2e tests
- [x]  Restore cache
- [x]  Install deps for each sample ( If there is no cache ):
- [x] If test failed ---> Create artifacts for screenshots and videos (Will be report for Cypress in the future )

Untitled Diagram drawio (3)

Second pipe:
- [x] Stop previous workflows
- [x]  Create cache & Install deps for all samples ( If there is no cache or if cache changed ):
         - Cache yarn.lock
         - Cache node_modules
         - Cache Cypress

### CODE REVIEW

**Comment by ScriptedAlchemy** [comment]
is this ready to merge?

**Comment by PeaceDate** [comment]
> is this ready to merge?

Not really need some approve from our side.
Will be ready from tomorrow. I'll add you for review

File: lerna.json [hunk 1]
```diff
@@ -3,46 +3,7 @@
   "npmClient": "yarn",
   "useWorkspaces": true,
   "packages": [
-    "./basic-host-remote/*",
```
Comment (valorkin) [inline lerna.json:6]: revert this?

File: package.json [hunk 2]
```diff
@@ -11,47 +11,8 @@
   "private": true,
   "workspaces": {
     "packages": [
-      "./basic-host-remote/*",
```
Comment (valorkin) [inline package.json:14]: revert?

File: lerna.json (hunk 1 above)
Comment (PeaceDate) [reply lerna.json:6]: Fixed

File: package.json (hunk 2 above)
Comment (PeaceDate) [reply package.json:14]: Fixed

**Pull request merged**